### ISSUE TITLE
gmsh2adcirc.py is not updated to MeshFormat version 4.1 (current)

### ISSUE THREAD

**Issue by GaRyu** [pprodano/pputils#1]
I downloaded pputils from here, downloaded Gmsh from their homepage, and ran into the problem `ValueError: invalid literal for int() with base 10: '1686419 5036 1 0\n'`. The problem appears to be that Gmsh have updated the file version to 4.1 while the pputils script expects version 2 which is legacy.

My first reaction was to attempt to make some changes to gmsh2adcirc.py that to make it work with MeshFormat 4.1, but I soon realized that it was easier to modify the call to Gmsh, and then I realized that there are other issues so maybe it's better to just drop Gmsh altogether? The best fix would probably be to first read the MeshFormat version from the text file and then handle mesh files according to what version they were produced in or throw an error if the version is not supported. Documentation for the different possible formats can be found at http://gmsh.info/doc/texinfo/gmsh.html#MSH-file-format 

This command does output a file in the old legacy version 2 format.

 `gmsh -2 out.geo -format msh2` 

MeshFormat 2.2 appears to work but gives a very rough division of the bounding box without any inclusion of nodes inside the bounding box. Therefore, I don't think MeshFormat 2.2 is working for pputils. MeshFormat 4.1 looks like an actual mesh of all the nodes but cannot be read by pputils. 

The output can be read by gmsh2adcirc.py without issues. However, using version msh22 is not particularly satisfying due to the very rough division of the mesh. I followed the manual from 2017 throughout the procedure. 

My raw data was 1.6 million nodes 50 meters apart or so (finer in some areas) of a river valley. The bounding box was a rectangle from each of the four corner nodes. Lines were along the waters edge and for some depth contour lines. I put none for holes.

**Comment by pprodano** [comment]
Hello,

You are right, I have not updated gmsh2adcirc.py to work with the latest Gmsh format. It is something I have considered doing, but just never found the time. As a matter of fact, I have generated multi-million node meshes and converted them without issue using the existing gmsh2adcirc.py.

I suspect there might be something else going on. If you wish, forward me your files, and I can take a look.

Closing the issue.
